cpp_exercises/day2/date/date.cpp

```diff
@@ -137,3 +138,43 @@ void Date::fromJulianDay(int64_t julianDay) {
 void Date::addDays(int days) {
     fromJulianDay(toJulianDay() + days);
 }
+
+Date Date::fromString(const std::string& string) {
+    if(std::count(string.begin(), string.end(), ' ') != 0)
+        return Date();
+
+    if(std::count(string.begin(), string.end(), '-') != 2)
```

Comment:
Io qui sarei più esplicito, controllerei che i trattini siano esattamente all'indice 2 e 5 della stringa. Se poi ce ne sono di più la funzione fallisce più in basso, se ce ne sono di meno lo scopriamo subito.
Meno iteriamo su tutti i caratteri della stringa e meglio è.